src/components/PostDetails.tsx

```diff
@@ -1,118 +1,80 @@
-};
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { post } = useAppSelector(state => state.selectedPost);
+  const { loaded, hasError } = useAppSelector(state => state.comments);
 
-export const PostDetails: React.FC<Props> = ({ post }) => {
   const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
+  useEffect(() => {
     setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
 
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post) {
+      dispatch(commentsActions.getComments(post.id)).then(res => {
+        setComments(res.payload as Comment[]);
+      });
     }
-  };
-
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  }, [post, dispatch]);
 
   const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
+    if (post) {
+      const newComment: Omit<Comment, 'id'> = {
         name,
         email,
         body,
         postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      };
+
+      await dispatch(commentsActions.createNewComment(newComment)).then(res =>
+        setComments(currentComments => [
+          ...currentComments,
+          res.payload as Comment,
+        ]),
+      );
     }
   };
 
   const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    // eslint-disable-next-line max-len
     setComments(currentComments =>
       currentComments.filter(comment => comment.id !== commentId),
     );
 
-    await commentsApi.deleteComment(commentId);
+    await dispatch(commentsActions.deleteAPIComment(commentId));
```

Comment:
Consider handling errors for the `deleteAPIComment` action. This ensures that any issues during the deletion process are managed appropriately.